containers/src/Kokkos_DynRankView.hpp

```diff
@@ -2170,7 +2170,7 @@ inline void realloc(DynRankView<T, P...>& v,
   const std::string label = v.label();
 
   v = drview_type();  // Deallocate first, if the only view to allocation
-  v = drview_type(label, n0, n1, n2, n3, n4, n5, n6);
+  v = drview_type(label, n0, n1, n2, n3, n4, n5, n6, n7);
```

Comment:
We need a unit test which catches this mistake.

Reply:
Allowing `DynRankView` to use 8 dimensions seems like a bigger change and should be in a different pull request if we decide we want it.
I am happy to just fix this issue here, though, and add a small test.

Reply:
I think DynRankView reserves the 8th rank for Sacado

Reply:
Are you saying the correct approach would be to restrict this constructor to seven dimensions as well?